pkg/handlers/generic/mutation/extraapiservercertsans/inject_test.go

```diff
@@ -37,39 +44,108 @@ var _ = Describe("Generate Extra API server certificate patches", func() {
 		return mutation.NewMetaGeneratePatchesHandler("", cl, NewPatch()).(mutation.GeneratePatches)
 	}
 
-	testDefs := []capitest.PatchTestDef{
+	testDefs := []testObj{
 		{
-			Name: "unset variable",
+			patchTest: capitest.PatchTestDef{
+				Name: "extra API server cert SANs set with AWS",
+				Vars: []runtimehooksv1.Variable{
+					capitest.VariableWithValue(
+						clusterconfig.MetaVariableName,
+						v1alpha1.ClusterConfigSpec{
+							GenericClusterConfig: v1alpha1.GenericClusterConfig{
+								ExtraAPIServerCertSANs: v1alpha1.ExtraAPIServerCertSANs{
+									"a.b.c.example.com",
+									"a.b.c.example.com",
+									"d.e.f.example.com",
+								},
+							},
+							AWS: &v1alpha1.AWSSpec{},
+						},
+					),
+				},
+				RequestItem: request.NewKubeadmControlPlaneTemplateRequestItem(""),
+				ExpectedPatchMatchers: []capitest.JSONPatchMatcher{{
+					Operation: "add",
+					Path:      "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration",
+					ValueMatcher: gomega.HaveKeyWithValue(
+						"apiServer",
+						gomega.HaveKeyWithValue(
+							"certSANs",
+							[]interface{}{"a.b.c.example.com", "d.e.f.example.com"},
+						),
+					),
+				}},
+			},
+			cluster: clusterv1.Cluster{
```

Comment:
nit: we should move this into a helper function.

Reply:
lets leave this alone for now. the testing story might be better addressed as a different PR